packages/wallet/src.ts/index.ts

```diff
@@ -1,266 +1,338 @@
+            } else {
+                // A lot of common tools do not prefix private keys with a 0x (see: #1166)
+                if (typeof (identity) === "string") {
+                    if (identity.match(/^[0-9a-f]*$/i) && identity.length === 64) {
+                        identity = "0x" + identity;
+                    }
+                }
+
+                const signingKey = new SigningKey(identity);
+                defineReadOnly(this, "_signingKey", () => signingKey);
+            }
+
+            defineReadOnly(this, "_mnemonic", (): Mnemonic => null);
+            defineReadOnly(this, "alias", computeAlias(this._signingKey().privateKey));
+        }
+        /* istanbul ignore if */
+        if (provider && !Provider.isProvider(provider)) {
+            logger.throwArgumentError("invalid provider", "provider", provider);
+        }
+
+        defineReadOnly(this, "provider", provider || null);
+    }
+
+    get mnemonic(): Mnemonic {
+        return this._mnemonic();
+    }
+
+    get privateKey(): string {
+        return this._signingKey().privateKey;
+    }
+
+    get publicKey(): string {
+        return this._signingKey().publicKey;
+    }
+
+    getAddress(): Promise<string> {
+        return Promise.resolve(this.address);
+    }
+
+    getAccount(): Promise<Account> {
+        return Promise.resolve(this.account);
+    }
+
+    getAlias(): Promise<string> {
+        return Promise.resolve(this.alias);
+    }
+
+    connect(provider: Provider): Wallet {
+        return new Wallet(this, provider);
+    }
+
+    connectAccount(accountLike: AccountLike): Wallet {
+        const eoa = {
+            privateKey: this._signingKey().privateKey,
+            address: getAddressFromAccount(accountLike),
+            alias: this.alias,
+            mnemonic: this._mnemonic()
+        };
+        return new Wallet(eoa, this.provider);
+    }
+
+    /**
+     * Signs a transaction with the key given upon creation.
+     * The transaction can be:
+     * - FileCreate - when there is only `fileChunk` field in the `transaction.customData` object
+     * - FileAppend - when there is both `fileChunk` and a `fileId` fields
+     * - ContractCreate - when there is a `bytecodeFileId` field
+     * - ContractCall - when there is a `to` field present. Ignores the other fields
+     *
+     * @param transaction - the transaction to be signed.
+     */
+    signTransaction(transaction: TransactionRequest): Promise<string> {
+        let tx: Transaction;
+        let arrayifiedData :Uint8Array;
+        if (transaction.data) {
+            arrayifiedData = arrayify(transaction.data);
+        }
+        const gas = BigNumber.from(transaction.gasLimit).toNumber();
+        if (transaction.to) {
+            tx = new ContractExecuteTransaction()
+                .setContractId(ContractId.fromSolidityAddress((transaction.to.toString())))
+                .setFunctionParameters(arrayifiedData)
```

Comment:
nit: we can do `transaction.data ? arrayify(transaction.data) : []` and skip the if above